linera-rpc/src/conversions.rs

```diff
@@ -52,23 +52,64 @@ impl From<ProtoConversionError> for Status {
     }
 }
 
-impl From<linera_base::VersionInfo> for grpc::VersionInfo {
-    fn from(version_info: linera_base::VersionInfo) -> grpc::VersionInfo {
+impl From<linera_version::CrateVersion> for grpc::CrateVersion {
+    fn from(
+        linera_version::CrateVersion {
+            major,
+            minor,
+            patch,
+        }: linera_version::CrateVersion,
+    ) -> Self {
+        Self {
+            major,
+            minor,
+            patch,
+        }
+    }
+}
+
+impl From<grpc::CrateVersion> for linera_version::CrateVersion {
+    fn from(
+        grpc::CrateVersion {
+            major,
+            minor,
+            patch,
+        }: grpc::CrateVersion,
+    ) -> Self {
+        Self {
+            major,
+            minor,
+            patch,
+        }
+    }
+}
+
+impl From<linera_version::VersionInfo> for grpc::VersionInfo {
+    fn from(version_info: linera_version::VersionInfo) -> grpc::VersionInfo {
         grpc::VersionInfo {
-            crate_version: version_info.crate_version.into(),
+            crate_version: Some(version_info.crate_version.into()),
             git_commit: version_info.git_commit.into(),
+            git_dirty: version_info.git_dirty,
             rpc_hash: version_info.rpc_hash.into(),
             graphql_hash: version_info.graphql_hash.into(),
             wit_hash: version_info.wit_hash.into(),
         }
     }
 }
 
-impl From<grpc::VersionInfo> for linera_base::VersionInfo {
-    fn from(version_info: grpc::VersionInfo) -> linera_base::VersionInfo {
-        linera_base::VersionInfo {
-            crate_version: version_info.crate_version.into(),
+impl From<grpc::VersionInfo> for linera_version::VersionInfo {
+    fn from(version_info: grpc::VersionInfo) -> linera_version::VersionInfo {
+        linera_version::VersionInfo {
+            crate_version: version_info
+                .crate_version
+                .unwrap_or(grpc::CrateVersion {
```

Comment:
We could also fail, I suppose.

Reply:
We're currently documenting optional fields with `optional` in our protobuf files.